components/text/dwdiff/Makefile

```diff
@@ -0,0 +1,50 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2024 Matthew R. Trower
+#
+
+BUILD_BITS=64
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=         dwdiff
+COMPONENT_VERSION=      2.1.4
+COMPONENT_SUMMARY=      dwdiff - a diff program that operates at the word level
+COMPONENT_PROJECT_URL=  https://os.ghalkes.nl/dwdiff.html
+COMPONENT_FMRI=         text/dwdiff
+COMPONENT_CLASSIFICATION=System/Text Tools
+COMPONENT_SRC=          $(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=      $(COMPONENT_SRC).tar.bz2
+COMPONENT_ARCHIVE_URL=  https://os.ghalkes.nl/dist/$(COMPONENT_ARCHIVE)
+COMPONENT_SIG_URL=      $(COMPONENT_ARCHIVE_URL).sig
+COMPONENT_ARCHIVE_HASH= \
+	sha256:df16fec44dcb467d65a4246a43628f93741996c1773e930b90c6dde22dd58e0a
+COMPONENT_LICENSE=      GPLv3
+
+TEST_TARGET=$(NO_TESTS)
+
+include $(WS_MAKE_RULES)/common.mk
+
+CONFIGURE_OPTIONS+=	INSTALL=/usr/ucb/install
+CONFIGURE_OPTIONS+=	CFLAGS=-D_POSIX_PTHREAD_SEMANTICS
```

Comment:
I recommend to add comment for the above two lines to explain why they are needed.  It is not obvious.  Thank you.

Reply:
Okay, `getpwuid_r(3C)` explains `_POSIX_PTHREAD_SEMANTICS`, but it is still unclear why we do need `/usr/ucb/install`.  What standards the `/usr/ucb/install` conforms to?  What is wrong with the default `/usr/gnu/bin/install` or `/usr/sbin/install`?

Reply:
`/usr/sbin/install`  uses very different flags from other Unixes.  It does not work for this package.  install(8) recommends /usr/ucb/install for better compatibility with other Unix `install` tools, and it works fine here.

`/usr/gnu/bin/install` would probably work also, but it is not the default for `install` unless `/usr/gnu/bin` happens to be first in your path.  Does the default build environment have such a configuration?

I favor the ucb tool, as it is present in illumos-gate, but I guess it doesn't really matter which is used here, as long as both are present in a base install.

Reply:
> `/usr/gnu/bin/install` would probably work also, but it is not the default for `install` unless `/usr/gnu/bin` happens to be first in your path. Does the default build environment have such a configuration?

```
PATH = $(PATH.gnu)
```

Reply:
Please try `PATH = $(PATH.gnu)` instead of `INSTALL=/usr/ucb/install`.  If it works, then please use it.  We prefer to set `PATH` instead of pointing individual tools.  Thank you.

Reply:
> PATH = $(PATH.gnu)

In fact that does not work (the advice is fine given the information I provided, but upon investigation there are more specifics).

If you want the details, the configure script does the following:

1. use `$INSTALL`, if set
2. test `install`, use that if it passes
3. use `./install.sh` if `install` does not pass
* On SunOS specifically, testing `install` is bypassed (because it is known not to work and (quote) "because it is dangerous", and `./install.sh` is used instead.  (Therefore, adjusting `PATH` becomes irrelevant.)

Okay, so with the above logic, `./install.sh` will always be used unless I set `$INSTALL`.  But `./install.sh` is broken for `make install` due to the relative path used, and the fact that the project has sub-directories.  So the options are:

- Patch `./configure`, and possibly other files, which may or may not prove trivial, and may break if these files are ever updated upstream.
- Possibly, we could symlink `./install.sh` into any sub-directories.  This would probably work; it may break with upstream upgrades.
- Submit a bug report upstream and hope it gets fixed (not something I want to get involved in personally).
- Set `$INSTALL` to a compatible install program.

Setting `INSTALL=/usr/ucb/install` seems clean, simple and reasonable to me.  If for some reason you really don't like the ucb install tool (which is provided by illumos-gate for exactly this purpose), I could instead set `INSTALL=/usr/gnu/bin/install` or `INSTALL=ginstall`.

I realize the above may seem confusing; I'm happy to provide more info or clarification on points.